site/content/contribute/more-info/desktop/architecture/navigation.md

```diff
@@ -0,0 +1,76 @@
+- **did-start-navigation** is another renderer process event that will fire once the page has started navigating. We can use this event to perform any actions when a certain URL is visited.
+- **new-window** is an event that will fire when the user tries to open a new window or tab. This commonly will fire when the user clicks on a link marked `target=_blank`. We attach this listener using the `setWindowOpenHandler` and will allow us to `allow` or `deny` the opening as we desire.
+
+In our application, we define all of these listeners in the `webContentEvents` module, and we attach them whenever a new `webContents` object is create to make sure that all renderer processes are correctly secured and set up correctly.
+
+#### Handling New Windows
+Our new window handler will *deny* the opening of a new Electron window if any of the following cases are true:
+- **Malformed URL:** Depending on the case, it will outright ignore it (if the URL could not be parsed) or it will open the user's default browser if it is somehow invalid in another way.
+- **Untrusted Protocol:** If the URL does not match an allowed protocol (allowed protocols include `http`, `https` and any other protocol that was explicitly allowed by the user). 
+    - In this case, it will pop the dialog to ask the user if the protocol should be allowed, and if so will open the URL in the users default application that corresponds to that protocol.
+- **Unknown Site:** If the URL does not match the root of a configured server, it will always try to open the link in the users default browser.
+    - If the URL DOES match the root of a configured server, we still will deny the window opening for a few cases:
+        - If the URL matches the public files route (`/api/v4/public/files/*`)
+        - If the URL matches the image proxy route (`/api/v4/image/*`)
+        - If the URL matches the help route (`/help/*`)
+    - For these cases, we will open the link in the user's browser.
+- **Deep Link Case**: If the URL doesn't match any of the above routes, but is still a valid configured server, we will generally treat is as the deep link cause and will instead attempt to show the correct tab and navigate to the corresponding URL within the app.
+
+There are two cases where we do allow the application to open a new window:
+- If the URL matches the `devtools:` protocol, so that we can open the Chrome Developer Tools.
+- If the URL is a valid configured server URL that corresponds to the plugins route (`/plugins/*`). In these cases we allow a single popup per tab to be opened for certain plugins to do things like OAuth (eg. GitHub, JIRA)
+
+Any other case will be automatically denied for security reasons.
+
+#### Links within the same window
+By default, the Mattermost app marks any link external to its application as `target=_blank` so that the application doesn't try to open it in the same window. Any other links should therefore be internal to the application.
```

Comment:
```suggestion
By default, the Mattermost app marks any link external to its application as `target=_blank`, so that the application doesn't try to open it in the same window. Any other links should therefore be internal to the application.
```